### ISSUE TITLE
Problem with a IMAP resource stream

### ISSUE THREAD

**Issue by wujku** [ddeboer/imap#245]
When I tried move message from `INBOX` to `Later` mailbox, `$message->move($later);` has stream from `Later` mailbox and `imap_mail_move` tried move message 100 to yourself.

```
$inbox = $connection->getMailbox('INBOX');
$later = $connection->getMailbox('Later');
$status = $later->getStatus();
$message = $inbox->getMessage(100);

$message->move($later);
```

**Comment by Slamdunk** [comment]
😮 

I've reproduced the bug, this has major impacts because every `imap_*` call must have a reinitialized resource; hope to push a fix soon.

**Comment by wujku** [comment]
This can also affect other methods.
How do you want to solve this problem?

**Comment by Slamdunk** [comment]
Passing to `Ddeboer\Imap\ImapResource` the mailbox context and reinitializing it at every `getStream()` call.

This bug is present from the beginning of the library.

**Comment by Slamdunk** [comment]
Close by https://github.com/ddeboer/imap/pull/247

**Comment by Slamdunk** [comment]
Released in [1.0.7](https://github.com/ddeboer/imap/releases/tag/1.0.7)
